src/main/kotlin/io/github/rybalkinsd/kohttp/dsl/context/HttpContext.kt

```diff
@@ -81,62 +79,19 @@ sealed class HttpContext(private val method: Method = GET) : IHttpContext {
 
 }
 
-@HttpDslMarker
-class HeaderContext {
-    private val map: MutableMap<String, Any> = mutableMapOf()
-
-    infix fun String.to(v: Any) {
-        map[this] = v
-    }
-
-    fun cookie(init: CookieContext.() -> Unit) {
-        map["cookie"] = CookieContext().also(init).collect()
-    }
-
-    internal fun forEach(action: (k: String, v: Any) -> Unit) = map.forEach(action)
-}
-
-@HttpDslMarker
-class CookieContext {
-    private val cookies: MutableList<Pair<String, Any>> = mutableListOf()
-
-    infix fun String.to(v: Any) {
-        cookies += Pair(this, v)
-    }
-
-    internal fun collect(): String = cookies.joinToString(separator = "; ") { (k, v) -> "$k=$v"}
-}
-
-@HttpDslMarker
-class ParamContext {
-    private val map: MutableMap<String, Any> = mutableMapOf()
-
-    infix fun String.to(v: Any) {
-        map[this] = v
-    }
-
-    internal fun forEach(action: (k: String, v: Any) -> Unit) = map.forEach(action)
-}
-
 class HttpGetContext : HttpContext()
 class HttpHeadContext : HttpContext(method = Method.HEAD)
 class HttpPutContext: HttpPostContext(method = Method.PUT)
 class HttpPatchContext: HttpPostContext(method = Method.PATCH)
 
 open class HttpPostContext(method: Method = Method.POST): HttpContext(method) {
-    var body: RequestBody = RequestBody.create(null, byteArrayOf())
+    private var body: RequestBody = RequestBody.create(null, byteArrayOf())
 
-    fun body(init: BodyContext.() -> RequestBody) {
-        body = BodyContext().init()
+    fun body(contentType: String? = null, init: BodyContext.() -> RequestBody) {
```

Comment:
Content type could be declared both in body and in header. Need to check who wins in okhttp. @shtykh